src/Microsoft.Extensions.Localization/ResourceManagerStringLocalizerFactory.cs

```diff
@@ -217,6 +225,24 @@ protected virtual ResourceLocationAttribute GetResourceLocationAttribute(Assembl
             return assembly.GetCustomAttribute<ResourceLocationAttribute>();
         }
 
+        /// <summary>Gets a <see cref="RootNamespaceAttribute"/> from the provided <see cref="Assembly"/>.</summary>
+        /// <param name="assembly">The assembly to get a <see cref="RootNamespaceAttribute"/> from.</param>
+        /// <returns>The <see cref="RootNamespaceAttribute"/> associated with the given <see cref="Assembly"/>.</returns>
+        /// <remarks>This method is protected and virtual for testing purposes only.</remarks>
+        protected virtual RootNamespaceAttribute GetRootNamespaceAttribute(Assembly assembly)
+        {
+            return assembly.GetCustomAttribute<RootNamespaceAttribute>();
+        }
+
+        private string GetRootNamespace(Assembly assembly)
+        {
+            var rootNamespaceAttribute = GetRootNamespaceAttribute(assembly);
+
+            return rootNamespaceAttribute?.RootNamespace == null
+                ? new AssemblyName(assembly.FullName).Name
+                : rootNamespaceAttribute.RootNamespace;
```

Comment:
Completely up to you: `rootNamespaceAttribute?.RootNamespace ?? new AssemblyName(assembly.FullName).Name`

Reply:
👍